src/server/src/Server.cpp

```diff
@@ -132,6 +137,16 @@ bool Server::updateModule()
 
         set_state(State::PositionControl);
     }
+    /*
+    else if (state == State::SetPosTrackParam)
+    {
+        drdSetPosTrackParam(amax_, vmax_, jerk_);
+    }
+    */
+    else if (state == State::SetPosMoveParam)
+    {
+        drdSetPosMoveParam(amax_, vmax_, jerk_);
+    }
```

Comment:
Shouldn't you move the state again to the previous one after changing the parameters ?

Also, for safety reasons maybe is better not allowing the user change the underlying motion generation parameters while the robot is under control. I suggest changing the logic such that the RPC call prevents the user from doing so unless the state is `Idle`. In both cases, you should reset the state of the module to something else after setting the parameters, otherwise the module will be stuck on setting the motion parameters.

Reply:
for me it worked fine. maybe we want to take a look together at this point?

Reply:
Sorry but I didn't understand what you meant. At the moment, the code is still not moving back to the previous state after changing the parameters. If you leave it like that, the state machine will continue to set the parameters instead of doing what it was supposed to do before the user asked to change the parameters. What do you think?